src/bin/add/args.rs

```diff
@@ -202,10 +206,15 @@ impl Args {
                             prefix = self.get_upgrade_prefix(),
                             version = package.version
                         );
+
                         dependency = dependency.set_version(&v);
-                        if let Some(registry) = &self.registry {
-                            dependency = dependency.set_registry(registry);
-                        }
+                    }
+                    let manifest = get_manifest_from_path(&dep_path)?;
+
+                    dependency = dependency.set_available_features(manifest.features());
```

Comment:
What if you instead move this into `parse_crate_name_from_uri`?  The manifest is already parsed and you can easily support git repos as well (since its parsed for those too)

Reply:
good one!